From 5ec90430360b7fe766d86247e3afe2848bf78f0e Mon Sep 17 00:00:00 2001
From: Konrad Leszczynski <konrad.leszczynski@intel.com>
Date: Mon, 8 Feb 2016 08:48:20 +0000
Subject: [PATCH 2/5] usb: xhci: additional check in MINF method assertion

Adding an additional check to the acpi_device due to
a corner-case, in which the handle structure is not
initialized.

Changing the dev_dbg usage to dev_err in the
xhci_intel_need_disable_stall function.

Change-Id: If0d92eacf78e340bcbabcdc093b6a97769895840
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18943
Signed-off-by: Konrad Leszczynski <konrad.leszczynski@intel.com>
Reviewed-on: https://android.intel.com:443/469354
---
 drivers/usb/host/xhci-intel-cap.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/host/xhci-intel-cap.c b/drivers/usb/host/xhci-intel-cap.c
index 7b6faab..d30d982 100644
--- a/drivers/usb/host/xhci-intel-cap.c
+++ b/drivers/usb/host/xhci-intel-cap.c
@@ -65,8 +65,8 @@ int xhci_intel_need_disable_stall(struct xhci_hcd *xhci)
 
 	/* Get ACPI device */
 	acpi_dev = ACPI_COMPANION(&pdev->dev);
-	if (!acpi_dev) {
-		dev_dbg(&pdev->dev, "No ACPI device!\n");
+	if (!acpi_dev || !acpi_dev->handle) {
+		dev_err(&pdev->dev, "No ACPI device!\n");
 		return 0;
 	}
 
@@ -74,7 +74,7 @@ int xhci_intel_need_disable_stall(struct xhci_hcd *xhci)
 	modem_status = acpi_evaluate_object(acpi_dev->handle, "MINF",
 			NULL, &buffer);
 	if (ACPI_FAILURE(modem_status)) {
-		dev_dbg(&pdev->dev, "No MINF method!\n");
+		dev_err(&pdev->dev, "No MINF method!\n");
 		return 0;
 	}
 
-- 
1.9.1

